src/sentry/utils/kvstore/abstract.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from datetime import timedelta
+from typing import Generic, Iterator, Optional, Sequence, Tuple, TypeVar
+
+
+K = TypeVar("K")
+V = TypeVar("V")
+
+
+class KVStorage(ABC, Generic[K, V]):
```

Comment:
This is generic mainly to allow returning various value types. I'd expect keys to generally be strings but I don't think there's a reason to constrain that behavior.